### ISSUE TITLE
All Requests Return `Invalid API key`

### ISSUE THREAD

**Issue by hstrowd** [honeybadger-io/honeybadger-webpack#3]
I recently tried adding this plugin to my webpack build and have triple checked my configurations (including verifying that I'm able to upload them manually via curl) but I'm still getting an `Invalid API key` error response for each source map upload attempt using this plugin.

I noticed you're using the `request` package for the HTTP communication in this plugin, so I added the `NODE_DEBUG=request` variable to my NPM build command, producing the following output:

```
REQUEST { uri: 'https://api.honeybadger.io/v1/source_maps ',
  callback: [Function],
  method: 'POST' }
REQUEST make request https://api.honeybadger.io/v1/source_maps
...
REQUEST onRequestResponse https://api.honeybadger.io/v1/source_maps 401 { server: 'nginx',
  date: 'Wed, 27 Dec 2017 14:47:48 GMT',
  'content-type': 'application/json; charset=utf-8',
  'transfer-encoding': 'chunked',
  connection: 'close',
  status: '401 Unauthorized',
  'x-frame-options': 'SAMEORIGIN',
  'x-xss-protection': '1; mode=block',
  'x-content-type-options': 'nosniff',
  'x-request-id': '276e4a84-f3ce-401e-8970-a0571de1a208',
  'x-runtime': '0.047786',
  'strict-transport-security': 'max-age=31536000' }
REQUEST reading response's body
REQUEST finish init function https://api.honeybadger.io/v1/source_maps
REQUEST response end https://api.honeybadger.io/v1/source_maps 401 { server: 'nginx',
  date: 'Wed, 27 Dec 2017 14:47:48 GMT',
  'content-type': 'application/json; charset=utf-8',
  'transfer-encoding': 'chunked',
  connection: 'close',
  status: '401 Unauthorized',
  'x-frame-options': 'SAMEORIGIN',
  'x-xss-protection': '1; mode=block',
  'x-content-type-options': 'nosniff',
  'x-request-id': '276e4a84-f3ce-401e-8970-a0571de1a208',
  'x-runtime': '0.047786',
  'strict-transport-security': 'max-age=31536000' }
``` 

I would have expected the initial request output to include the form data. Is this an issue you've heard about previously or run across thus far?

I have added this plugin to the bottom of the `plugins` array in my webpack config but it seems like some of the asset compilation steps are running asynchronously, resulting in this plugin getting invoked before the asset creation has completed and causing the form data not to be ready by the time these source map upload requests are triggered. I saw that the `form-data` package was being used to generate the request form payload (https://github.com/honeybadger-io/honeybadger-webpack/blob/master/src/HoneybadgerSourceMapPlugin.js#L90) and wondered if this could be opening the potential for the request to be triggered without this data being included. The request package documentation includes the following statement in relation to using the form-data instance directly: `This can be modified until the request is fired on the next cycle of the event-loop. ` (see https://www.npmjs.com/package/request#forms).

Don't hesitate to let me know if any additional information is needed to help debug this issue. Thank you very much for your help.

**Comment by monde** [comment]
@hstrowd - thanks for the detailed debug feedback ... are you certain you are setting the `apiKey` correctly when you initialize the `HoneybadgerSourceMapPlugin`? The only way I've been able to reproduce `401 Unauthorized` / `{"error":"Invalid API key"}` in my sample environment with curl and also with my webpack environment is if I deliberately set the `apiKey` to an invalid value. The project's API key can be found in your project's panel on Honeybadger under Settings / Api Key. Let me know.

**Comment by hstrowd** [comment]
@monde Thanks so much for the quick reply.

tl;dr: You were correct. I should have trusted the error message more than my debugging output. Sorry for the confusion.

At the root of the issue was the fact that I was pulling the API key attribute from my .env file using the `dotenv-webpack` plugin, which unbeknownst to me was including a set of `"`s around the value itself. As a result when I printed this content to the console I must have just ignored these characters entirely. Again, sorry for the mixup and wasting your time. Thanks again for your help getting to the bottom of it.

**Comment by monde** [comment]
I learned some things from your debug analysis so it was worthwhile to me!
